mfx/optimize/align.py

```diff
@@ -1,52 +1,96 @@
+from typing import Literal
+from pydantic import validate_call, ConfigDict
 from bluesky import RunEngine
 
 
+Diagnostics = Literal["xcs1", "dg1", "dg2"]
+Methods = Literal["xopt", "blop"]
+Devices = Literal["yag", "wave8"]
+Turbo = Literal["safety", "optimize"]
+
+
+def validate_w_lowercase_args(func):
+    """
+   Decorator to make string inputs lowercase before validating.
+
+    Parameters:
+    -----------
+    func (Callable): 
+        The function to decorate.
+
+    Returns:
+    --------
+    Callable: 
+        The decorated function with string arguments converted to lowercase.
+    """
+    def wrapper(*args, **kwargs):
+        # Convert all string arguments to lowercase
+        new_args = tuple(arg.lower() if isinstance(arg, str) else arg for arg in args)
+        new_kwargs = {k: v.lower() if isinstance(v, str) else v for k, v in kwargs.items()}
+
+        # Call the original function with the modified arguments, validated by Pydantic
+        validated_func = validate_call(func, config=ConfigDict(validate_default=True))
+        return validated_func(*new_args, **new_kwargs)
+
+    return wrapper
+
+
 class Align:
-    def __init__(self):
-        self.beam_alignment_diagnostics = ["XCS1", "DG1", "DG2"]
-        self.beam_alignment_methods = ["Xopt", "blop"]
-        self.beam_alignment_devices = ["yag", "wave8"]
+    @validate_call
+    def __init__(self, mirror_pitch: list[float] = [-549.0, -546.0]):
+        self.mirror_pitch: list[float] = mirror_pitch
 
+    @validate_w_lowercase_args
     def beam(
             self,
             with_goal: float,
-            on_diagnostic: str = "DG1",
-            with_method: str = 'Xopt',
-            using_device: str = "yag",
-            xopt_turbo_option: str = "safety",
+            on_diagnostic: Diagnostics = "dg1",
+            with_method: Methods = "xopt",
+            using_device: Devices = "yag",
+            xopt_turbo_option: Turbo = "safety",
+            xopt_rand_evaluate: int = 3,
+            xopt_steps: int = 10,
+            blop_qr_n: int = 16,
+            blop_qei_n: int = 16,
+            blop_qei_iterations: int = 5
             ):
         """Perform Beam Alignment
 
-        Parameters:
-
-            on_diagnostic (str): diagnostic to use for alignment. Options: "XCS1, DG1, DG2".
-
-            with_method(str): method to use for alignment. Options: "blop, Xopt".
+        Parameters
+        ----------
+        with_goal : float
+            Goal to align to.
+        on_diagnostic : str, optional
+            Diagnostic to use for alignment. Options: "XCS1, DG1, DG2". Default is "dg1".
+        with_method : str, optional
+            Method to use for alignment. Options: "blop, Xopt". Default is "xopt".
```

Comment:
Nitpick: if we're forcing everything lowercase, let's be consistent in the docstring so we don't accidentally have people putting capital strings in when they don't need to.

```suggestion
            Diagnostic to use for alignment. Options: "xcs1, dg1, dg2". Default is "dg1".
        with_method : str, optional
            Method to use for alignment. Options: "blop, xopt". Default is "xopt".
```